src/users/users.ts

```diff
@@ -0,0 +1,135 @@
+import { WorkOS } from '../workos';
+import {
+  AddUserToOrganizationOptions,
+  AuthenticateManagedUserOptions,
+  AuthenticateUnmanagedUserOptions,
+  AuthenticationResponse,
+  CompletePasswordResetOptions,
+  CreateEmailVerificationChallengeOptions,
+  CreateEmailVerificationChallengeResponse,
+  CreatePasswordResetChallengeOptions,
+  CreatePasswordResetChallengeResponse,
+  CreateUnmanagedUserOptions,
+  ListUsersOptions,
+  RemoveUserFromOrganizationOptions,
+  RevokeSessionOptions,
+  User,
+  VerifySessionOptions,
+  VerifySessionResponse,
+} from './interfaces';
+import { List } from '../common/interfaces/list.interface';
+
+export class Users {
+  constructor(private readonly workos: WorkOS) {}
+
+  async getUser(userId: string): Promise<User> {
+    const { data } = await this.workos.get(`/users/${userId}`);
+    return data;
+  }
+
+  async listUsers(options?: ListUsersOptions): Promise<List<User>> {
+    const { data } = await this.workos.get('/users', {
+      query: options,
+    });
+    return data;
+  }
+
+  async createUnmanagedUser(
+    payload: CreateUnmanagedUserOptions,
+  ): Promise<User> {
+    const { data } = await this.workos.post('/users', payload);
+
+    return data;
+  }
+
+  async authenticateUnmanagedUser(
```

Comment:
Perhaps be more specific with the name:

```suggestion
  async authenticateUserWithPassword(
```

Because OAuth users are unmanaged but don't go through this endpoint.

Reply:
done.